store/cachekv/store_test.go

```diff
@@ -684,3 +687,21 @@ func BenchmarkCacheKVStoreGetKeyFound(b *testing.B) {
 		st.Get([]byte{byte((i & 0xFF0000) >> 16), byte((i & 0xFF00) >> 8), byte(i & 0xFF)})
 	}
 }
+
+//--------------------------------------------------------
+
+func BenchmarkCacheKVStoreSetAndCommit(b *testing.B) {
+	mem := dbadapter.Store{DB: dbm.NewMemDB()}
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		store := cachekv.NewStore(mem)
+		store1 := store.Clone()
+		for j := 0; j < 10; j++ {
+			store1.Set(sdk.Uint64ToBigEndian(uint64(i+j)), []byte{byte(i)})
+		}
+		store.Restore(store1)
+		store.Write()
```

Comment:
This benchmark allocates a new store, clones that store, sets a few values in the clone, restores the initial store from the clone, and the persists the initial store to an underlying persistence layer, with each iteration of b.N. That's an enormous amount of work which is dominated by allocations. You're not getting any meaningful information out of this benchmark.

Reply:
Just simulate a common path on real world case, two layers of cache store, a bunch of writes, and commit

Reply:
That's fine, but if you do that work as part of a benchmark, the benchmark isn't providing any useful information, because that "common path on real world case" is dominated by external factors unrelated to the code in the package.

Reply:
It shows this code path get faster or slower, if it can be reproduced stablely, it must be because of the code changes in this repo.

Reply:
It does not show that, not really, no. The benchmark is measuring allocation performance, whatever changes are made to the code in the package are super tiny in comparison to the other work happening.

Reply:
Do you mean allocation speed can increase 36% stablely and magically for reasons unrelated to the code changes?

Reply:
I believe this benchmark is showing the performance of a common pattern used in the SDK, which is pretty useful, even if it is influenced heavily by external factors in a production environment, having insight into this is still useful.

Reply:
This is pretty simple, the code in the b.N loop is allocating a new store, cloning it, and then calling restore, in each iteration. The overhead of those things dominate whatever you're trying to measure here.